### DISCUSSION TITLE
Building 108.0.5359.156 ld.lld: error: unable to find library -latomic

### DISCUSSION THREAD

**Discussion by BlueSky15171** [bromite/bromite#2628] [Builds]
ninja -j8 -C out/arm system_webview_apk
ninja: Entering directory `out/arm'
[4/23588] LINK ./md5sum_bin
FAILED: md5sum_bin exe.unstripped/md5sum_bin exe.unstripped/md5sum_bin.map.gz 
python3 "../../build/toolchain/gcc_link_wrapper.py" --output="./md5sum_bin" --strip="../../third_party/llvm-build/Release+Asserts/bin/llvm-strip" --unstripped-file="./exe.unstripped/md5sum_bin" --map-file "./exe.unstripped/md5sum_bin.map.gz" -- ../../third_party/llvm-build/Release+Asserts/bin/clang++ -fuse-ld=lld -Wl,--fatal-warnings -Wl,--build-id=sha1 -fPIC -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now -Wl,--icf=all -Wl,--color-diagnostics -Wl,-mllvm,-instcombine-lower-dbg-declare=0 -flto=thin -Wl,--thinlto-jobs=all -Wl,--thinlto-cache-dir=thinlto-cache -Wl,--thinlto-cache-policy=cache_size=10\%:cache_size_bytes=40g:cache_size_files=100000 -Wl,-mllvm,-import-instr-limit=5 -march=armv7-a -Wl,--no-rosegment -Wl,--no-call-graph-profile-sort -Wl,--exclude-libs=libvpx_assembly_arm.a --unwindlib=none --target=arm-linux-androideabi23 -no-canonical-prefixes -Wl,--warn-shared-textrel -Wl,-O2 -Wl,--gc-sections -Wl,-z,defs -Wl,--as-needed -nostdlib++ --sysroot=../../third_party/android_ndk/toolchains/llvm/prebuilt/linux-x86_64/sysroot -Wl,--warn-shared-textrel -Wl,--lto-O0 -fsanitize=cfi-vcall -fsanitize=cfi-derived-cast -fsanitize=cfi-unrelated-cast -pie -Bdynamic -Wl,-z,nocopyreloc -Wl,-wrap,calloc -Wl,-wrap,free -Wl,-wrap,malloc -Wl,-wrap,memalign -Wl,-wrap,posix_memalign -Wl,-wrap,pvalloc -Wl,-wrap,realloc -Wl,-wrap,valloc -Wl,-wrap,malloc_usable_size -Wl,-wrap,realpath -Wl,-wrap,strdup -Wl,-wrap,strndup -Wl,-wrap,getcwd -Wl,-wrap,asprintf -Wl,-wrap,vasprintf -o "./exe.unstripped/md5sum_bin" -Wl,--start-group @"./md5sum_bin.rsp"  -Wl,--end-group  -landroid_support -ldl -lm -landroid -llog -latomic ../../base/android/library_loader/anchor_functions.lds
ld.lld: error: unable to find library -latomic
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
[11/23588] CXX obj/ui/events/platform/platform/platform_event_source.o
ninja: build stopped: subcommand failed.

------------

how fixed this question?
Thanks.

**Comment by SergeyDjam** [comment]
install libatomic-devel

**Reply by BlueSky15171** [reply]
Thanks. 
I try install it ,but apt not found libatomic-devl on Ubuntu20.04, found libatomic1.The libatomic I installed is invalid.

**Reply by BlueSky15171** [reply]
i fixed , use `use_sysroot=true`

**Comment by uazo** [comment]
technically the reason is that sysroot is disabled in gn but in android it doesn't seem to have any purpose.
```
use_sysroot
    Current value = false
      From /home/lg/working_dir/bromite/build/bromite.gn_args:45
    Overridden from the default = true
      From //build/config/sysroot.gni:22

    Controls default is_linux sysroot. If set to true, and sysroot
    is empty, default sysroot is calculated.
```

**Reply by BlueSky15171** [reply]
Thanks.
i fixed. use `use_sysroot=true`
